### PR TITLE
wqueue: update wqueue to support using wqueue before wqueue start

### PR DESCRIPTION
**Pull request by GUIDINGLI**
## Summary
wqueue: update wqueue to support using wqueue before wqueue start

## Impact
wqueue

## Testing
VELA

### CODE REVIEW

**Comment by Ouss4** [comment]
@xiaoxiang781216 @GUIDINGLI how is the work queue used before starting?

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 @GUIDINGLI how is the work queue used before starting?

up_initialize and board_early_initialize get called before nx_workqueues:
https://github.com/apache/incubator-nuttx/blob/master/sched/init/nx_start.c#L682-L699
Some driver initialize in up_initialize or board_early_initialize may schedule work to the worker thread.

**Review by Ouss4 — APPROVED** [review]
LGTM!
Please remove the gerrit id to pass the CI check.

**Review by pkarashchenko — APPROVED** [review]
Please remove `Change-Id: Ie91d8e71ea0191cff62118071398b3f3fd4294f1` from commit message and force-push commit

**Comment by hartmannathan** [comment]
Does it matter that `nxsem_set_protocol(&wqueue->sem, SEM_PRIO_NONE);` is still done in `work_thread_create()`?

**Comment by GUIDINGLI** [comment]
> Does it matter that `nxsem_set_protocol(&wqueue->sem, SEM_PRIO_NONE);` is still done in `work_thread_create()`?

add new MARCO `NXSEM_INITIALIZER`, to handle this

File: include/nuttx/semaphore.h [hunk 1]
```diff
@@ -36,6 +36,21 @@
...
+#ifdef CONFIG_PRIORITY_INHERITANCE
+# if CONFIG_SEM_PREALLOCHOLDERS > 0
+#  define NXSEM_INITIALIZER(c, f) \
+    {(c), (f), NULL}               /* semcount, flags, hhead */
+# else
+#  define NXSEM_INITIALIZER(c, f) \
+    {(c), (f), {SEMHOLDER_INITIALIZER, SEMHOLDER_INITIALIZER}} /* semcount, flags, holder[2] */
+# endif
+#else
+#  define NXSEM_INITIALIZER(c, f) \
```
Comment (xiaoxiang781216) [inline include/nuttx/semaphore.h:50]: let's define:
```
SEM_INITIALIZER(c) NXSEM_INITIALIZER(c, 0)
```

File: include/nuttx/semaphore.h [hunk 2]
```diff
@@ -36,6 +36,21 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
+/* Initializers */
+
+#ifdef CONFIG_PRIORITY_INHERITANCE
+# if CONFIG_SEM_PREALLOCHOLDERS > 0
+#  define NXSEM_INITIALIZER(c, f) \
+    {(c), (f), NULL}               /* semcount, flags, hhead */
```
Comment (xiaoxiang781216) [inline include/nuttx/semaphore.h:44]: align with the comment at line 47 and 51

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**